crates/wasm-smith/src/core.rs

```diff
@@ -328,10 +328,16 @@ enum DataSegmentKind {
     Passive,
     Active {
         memory_index: u32,
-        offset: Instruction,
+        offset: ConstExpr,
     },
 }
 
+#[derive(Debug)]
+pub(crate) enum GlobalInitExpr {
+    FuncRef(u32),
+    ConstExpr(ConstExpr),
+}
```

Comment:
I don’t love this, but there was code out there that was matching on `Instruction::FuncRef` before and adding a type like this seemed like the path of least resistance.